torch/__init__.py

```diff
@@ -33,61 +33,28 @@
 # Load the extension module
 ################################################################################
 
-# Loading the extension with RTLD_GLOBAL option allows to not link extension
-# modules against the _C shared object. Their missing THP symbols will be
-# automatically filled by the dynamic loader.
-import os as _dl_flags
-
-# if we have numpy, it *must* be imported before the call to setdlopenflags()
-# or there is risk that later c modules will segfault when importing numpy
-try:
-    import numpy as _np  # noqa: F401
-except ImportError:
-    pass
-
 if platform.system() == 'Windows':
-    # first get nvToolsExt PATH
-    def get_nvToolsExt_path():
-        NVTOOLEXT_HOME = _dl_flags.getenv('NVTOOLSEXT_PATH', 'C:\\Program Files\\NVIDIA Corporation\\NvToolsExt')
+    NVTOOLSEXT_PATH = os.getenv('NVTOOLSEXT_PATH', 'C:\\Program Files\\NVIDIA Corporation\\NvToolsExt')
```

Comment:
@peterjc123 Do we really need this code now? We're no longer using `ctypes` to dlopen any libraries -- they should all get listed as dependencies of libtorch.

Reply:
<del> But the dynamic library `torch.dll` relies on `nvtoolext64_1.dll`, right? In that case, we still need it because the environmental variable is not reliable from the user's side. </del> We can package that dll into the package so that we don't really need this code.

Reply:
Can you elaborate? Why would they have issues?

Reply:
Because:
1. The directory of the nvtx library is not added into PATH on Windows. Windows relies on PATH to search for dynamic libraries.
2. All the dynamic libraries are eagerly loaded (not lazily) unless `/DELAYLOAD` is specified on the linker line. https://docs.microsoft.com/en-us/cpp/build/reference/delayload-delay-load-import?view=vs-2019
3. The following dependency chain exists: `_C -> torch_python -> nvtx`, so I don't think it's only a dependent of libtorch. 
https://github.com/pytorch/pytorch/pull/28536/files#diff-2eeaed663bd0d25b7e608891384b7298L613 https://github.com/pytorch/pytorch/pull/28536/files#diff-94eab65863273226cd6b960d685a427eR179

When users compile PyTorch by themselves, they don't package the nvtx library along with the torch libraries. (which is done through the release scripts in pytorch/builder). Then they will experience the import errors because `nvtoolsext64_1.dll` could not be found at runtime.

Reply:
So basically what you're saying is that there's nothing like RPATH for Windows, and all the DLLs that you depend on should be in your PATH when you load a shared library?

Reply:
Yeah, that's what I mean. For more details, you may read [this](http://ibob.github.io/blog/2018/12/16/windows-rpath/) article.

Reply:
Actually, things are getting worse in Python 3.8. It doesn't read from `PATH` anymore, but os.add_dll_directory should be called manually instead. This means that we will need to add more code here, for finding cuda and cudnn DLLs. In order to make this file clean, maybe we should move this into a different script in the root directory.